### ISSUE TITLE
Link resolution failed

### ISSUE THREAD

**Issue by JaxsonWang** [markedjs/marked#1268]
see this [demo](https://marked.js.org/demo/?text=%5Bwd%5D(https%3A%2F%2Faaa.com%2F(wd)))

or this [demo](https://marked.js.org/demo/?text=%5Bdemo%5D(https%3A%2F%2Fwiki.archlinux.org%2Findex.php%2FSamba_(%E7%AE%80%E4%BD%93%E4%B8%AD%E6%96%87)%23.E6.B7.BB.E5.8A.A0.E7.94.A8.E6.88.B7))

If the link contains `()` it will fail to parse. Thanks for solving this problem.

**Comment by styfle** [comment]
Thanks for the bug report!

I confirmed that this is a bug in marked, and works fine in CommonMark.
- [working demo](https://spec.commonmark.org/dingus/?text=%5Bwd%5D(https%3A%2F%2Faaa.com%2F(wd)))
- [spec reference](https://spec.commonmark.org/0.28/#example-468)

**Comment by UziTech** [comment]
@Feder1co5oave wasn't #1135 supposed to fix this?

**Comment by Feder1co5oave** [comment]
Is this relative to master? I don't think that pr was released yet?
It should support up to one level of nested, balanced, parenthesis, yes.

**Comment by Feder1co5oave** [comment]
That demo uses the latest released version, the bug **is** fixed in master:
```html
$ bin/marked
[wd](https://aaa.com/(wd))

[demo](https://wiki.archlinux.org/index.php/Samba_(简体中文)#.E6.B7.BB.E5.8A.A0.E7.94.A8.E6.88.B7)

<p><a href="https://aaa.com/(wd)">wd</a></p>
<p><a href="https://wiki.archlinux.org/index.php/Samba_(%E7%AE%80%E4%BD%93%E4%B8%AD%E6%96%87)#.E6.B7.BB.E5.8A.A0.E7.94.A8.E6.88.B7">demo</a></p>
```

Please make sure the bug is still there before confirming it.

**Comment by styfle** [comment]
I confirmed this is fixed in master and will be released in 0.4.0
